packages/xod-client/src/user/reducer.js

```diff
@@ -0,0 +1,20 @@
+import { UPDATE_COMPILE_LIMIT } from './actionTypes';
+
+const userReducer = (state = {}, action) => {
+  switch (action.type) {
+    case UPDATE_COMPILE_LIMIT: {
+      const limit = action.payload;
+      return {
+        ...state,
+        limit: !limit ? null : {
+          ...limit,
+          left: Math.max(0, limit.limit - limit.pending - limit.used),
```

Comment:
🤔 
I think it will be better to move it into the selector (it could be calculated, so we don't need to store in the state).